www/front_src/src/Resources/Filter/Edit/EditFilterCard.tsx

```diff
@@ -0,0 +1,198 @@
+import * as React from 'react';
+
+import { makeStyles } from '@material-ui/core';
+import DeleteIcon from '@material-ui/icons/Delete';
+
+import {
+  ContentWithCircularLoading,
+  TextField,
+  IconButton,
+  useRequest,
+  ConfirmDialog,
+  useSnackbar,
+  Severity,
+} from '@centreon/ui';
+
+import { useFormik } from 'formik';
+import * as Yup from 'yup';
+import { isNil, not, all, equals, any } from 'ramda';
+
+import {
+  labelDelete,
+  labelAskDelete,
+  labelCancel,
+  labelFilterDeleted,
+  labelFilterUpdated,
+  labelName,
+  labelFilter,
+  labelNameCannotBeEmpty,
+} from '../../translatedLabels';
+import { updateFilter, deleteFilter } from '../api';
+import { Filter } from '../models';
+import { useResourceContext } from '../../Context';
+
+const useStyles = makeStyles((theme) => ({
+  filterCard: {
+    display: 'grid',
+    gridAutoFlow: 'column',
+    gridGap: theme.spacing(2),
+    alignItems: 'center',
+    gridTemplateColumns: '2fr 1fr',
+  },
+  filterNameInput: {},
+  filterEditActions: {
+    display: 'grid',
+    gridAutoFlow: 'column',
+    gridGap: theme.spacing(1),
+    justifyContent: 'flex-start',
+  },
+}));
+
+interface Props {
+  filter: Filter;
+}
+
+const EditFilterCard = ({ filter }: Props): JSX.Element => {
+  const classes = useStyles();
+
+  const {
+    setFilter,
+    filter: currentFilter,
+    loadCustomFilters,
+    customFilters,
+    sendingListCustomFiltersRequest,
+  } = useResourceContext();
+
+  const { showMessage } = useSnackbar();
+
+  const [deleting, setDeleting] = React.useState(false);
+
+  const {
+    sendRequest: sendUpdateFilterRequest,
+    sending: sendingUpdateFilterRequest,
+  } = useRequest({
+    request: updateFilter,
+  });
+
+  const {
+    sendRequest: sendDeleteFilterRequest,
+    sending: sendingDeleteFilterRequest,
+  } = useRequest({
+    request: deleteFilter,
+  });
+
+  const { name, id } = filter;
+
+  const validationSchema = Yup.object().shape({
+    name: Yup.string().required(labelNameCannotBeEmpty),
+  });
+
+  const form = useFormik({
+    initialValues: {
+      name,
+    },
+    validationSchema,
+    onSubmit: (values) => {
+      sendUpdateFilterRequest({ ...filter, name: values.name }).then(
+        (updatedFilter) => {
+          if (equals(updatedFilter.id, currentFilter.id)) {
+            setFilter(updatedFilter);
+          }
+
+          showMessage({
+            message: labelFilterUpdated,
+            severity: Severity.success,
+          });
+
+          loadCustomFilters();
+        },
+      );
+    },
+  });
+
+  const askDelete = (): void => {
+    setDeleting(true);
+  };
+
+  const confirmDelete = (): void => {
+    setDeleting(false);
+
+    sendDeleteFilterRequest(filter).then(() => {
+      showMessage({ message: labelFilterDeleted, severity: Severity.success });
+
+      loadCustomFilters();
+    });
+  };
+
+  const cancelDelete = (): void => {
+    setDeleting(false);
+  };
+
+  const loading = isNil(customFilters);
```

Comment:
```suggestion
  const customFiltersEmpty = isNil(customFilters);
```
This is not a loading state but this indicates that `customFilters` is null or undefined